src/modules/job-exec/job-exec.c

```diff
@@ -954,14 +954,16 @@ static int job_start (struct job_exec_ctx *ctx, const flux_msg_t *msg)
     flux_future_t *f = NULL;
     struct jobinfo *job;
 
+    /* If anything goes wrong before job->id is set, we cannot call
+     * jobinfo_fatal_error() because it's not possible to construct a
+     * valid response message for job-manager.  Just log the (unlikely) error.
+     */
     if (!(job = jobinfo_new ()))
         return -1;
 
     if (!(job->req = flux_msg_copy (msg, true))) {
         flux_log_error (ctx->h, "start: flux_msg_copy");
         jobinfo_decref (job);
-        if (flux_respond_error (ctx->h, msg, errno, "flux_msg_copy failed") < 0)
-            flux_log_error (ctx->h, "flux_respond_error");
         return -1;
```

Comment:
I thought we wanted to respond to job-manager with a "generic" (i.e. non-jobid response) when the job-exec module encounters a fatal error, so that the job-manager/exec protocol is "stopped" and the presumably internal error can be rectified before we encounter many failures.

For example, if the job-exec gets an invalid message from the job-manager, that probably isn't a one time thing, but may indicate a version incompatibility problem, etc.

Reply:
Oh you are right, I was confused.  The job manager is plumbed to tear down the exec system interface if it ever receives a "normal" error response to `start`.

There is still an issue because both error and success responses are being sent in the EPROTO case.  Let me rework, and fix the commit comment